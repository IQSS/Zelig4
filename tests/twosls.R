library(ZeligLeastSquares)

data(klein)

formula <- list(
                mu1 = C ~ Wtot + P1,
                mu2 = I ~ P + P1 + K1,
                mu3 = Wp ~ X + X1 + Tm,
                inst= ~ P1 + K1 + X1 + Tm + Wg + G
                )

z.out<-zelig(formula=formula, model="twosls",data=klein, cite=F)

x.out <-setx(z.out)

s.out <-sim(z.out,x=x.out)

summary(s.out)


# Plot

user.prompt()
plot(s.out)


