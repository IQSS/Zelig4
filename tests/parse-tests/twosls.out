## 
##  Zelig (Version 3.5.3, built: 2010-01-20)
##  Please refer to http://gking.harvard.edu/zelig for full
##  documentation or help.zelig() for help with commands and
##  models supported by Zelig.
##

##  Zelig project citations:
##    Kosuke Imai, Gary King, and Olivia Lau. (2009).
##    ``Zelig: Everyone's Statistical Software,''
##    http://gking.harvard.edu/zelig.
##  and
##    Kosuke Imai, Gary King, and Olivia Lau. (2008).
##    ``Toward A Common Framework for Statistical Analysis
##    and Development,'' Journal of Computational and
##    Graphical Statistics, Vol. 17, No. 4 (December)
##    pp. 892-913. 

##  To cite individual Zelig models, please use the citation format printed with
##  each model run and in the documentation.
##
$mu1
C ~ Wtot + P + P1

$mu2
I ~ P + P1 + K1

$mu3
Wp ~ X + X1 + Tm

$inst
~1
<environment: 0x100f21e68>

attr(,"systEqns")
[1] "mu1" "mu2" "mu3"
attr(,"ancilEqns")
[1] "inst"
attr(,"variables")
attr(,"variables")$mu1
list(C, Wtot, P, P1)

attr(,"variables")$mu2
list(I, P, P1, K1)

attr(,"variables")$mu3
list(Wp, X, X1, Tm)

attr(,"variables")$inst
list()

attr(,"term.labels")
attr(,"term.labels")$mu1
[1] "Wtot" "P"    "P1"  

attr(,"term.labels")$mu2
[1] "P"  "P1" "K1"

attr(,"term.labels")$mu3
[1] "X"  "X1" "Tm"

attr(,"term.labels")$inst
character(0)

attr(,"indVars")
[1] "Wtot" "P"    "P1"   "K1"   "X"    "X1"   "Tm"  
attr(,"depVars")
attr(,"depVars")$mu1
[1] "C"

attr(,"depVars")$mu2
[1] "I"

attr(,"depVars")$mu3
[1] "Wp"

attr(,"depFactors")
[1] FALSE
attr(,"constraints")
[1] FALSE
attr(,"subs")
[1] FALSE
attr(,"response")
[1] 1
attr(,"intercept")
attr(,"intercept")$mu1
[1] 1

attr(,"intercept")$mu2
[1] 1

attr(,"intercept")$mu3
[1] 1

attr(,"intercept")$inst
[1] 1

